test/functional/test_framework/wallet.py

```diff
@@ -84,8 +84,11 @@ def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_sp
         tx_hex = tx.serialize().hex()
 
         tx_info = from_node.testmempoolaccept([tx_hex])[0]
-        self._utxos.append({'txid': tx_info['txid'], 'vout': 0, 'value': send_value})
+        utxos.append({'txid': tx_info['txid'], 'vout': 0, 'value': send_value})
         from_node.sendrawtransaction(tx_hex)
         assert_equal(tx_info['vsize'], vsize)
         assert_equal(tx_info['fees']['base'], fee)
+        # update our internal utxo set after `sendrawtransaction` and the above assertions succeed to ensure a thrown exception
```

Comment:
I fixed this in #21762 by moving the `self._utxos.append` to the line after `sendrawtransaction`. Wdyt?

Reply:
Looks great, prefer you're implementation